msteams-platform/toolkit/provision.md

```diff
@@ -1,616 +1,436 @@
+    AAD_APP_ACCESS_AS_USER_PERMISSION_ID=<id of access_as_user permission> # example: 00000000-0000-0000-0000-000000000000
+    ```
 
-You can customize your bot or the Teams app by adding configuration snippets to use an Azure AD app created for your Teams app.
-Perform in the following ways to customize the Teams app:
+    1. If your application requires an Azure AD app client secret, add below environment variable and its value to `env\.env.{env}.user` file.
 
-* [Use an existing Azure AD app for your Teams app](#use-an-existing-azure-ad-app-for-your-teams-app)
-* [Use an existing Azure AD app for your bot](#use-an-existing-azure-ad-app-for-your-bot)
+    ```typescript
+    SECRET_AAD_APP_CLIENT_SECRET=<value of Azure AD application's client secret>
+    ```
 
-#### Use an existing Azure AD app for your Teams app
+    > [!NOTE]
+    >
+    > * Remember to update the environment variable names in the examples if you uses different names in `writeToEnvironmentFile`.
+    >
+    > * If you don't use `aadApp/create` action to create Azure AD application, you can add necessary environment variables with your preferred name without following above steps.
+    >
+    > * Ensure not to share the same Azure AD app in multiple environments. If you don't have permission to update the Azure AD app, you get a warning with instructions about how to manually update the Azure AD app. Follow the instructions to update your Azure AD app after provision.
 
-You can add the following configuration snippet to `.fx\configs\config.{env}.json` file to use an Azure AD app created for your Teams app. If you don't have an Azure AD app yet or you already have one but don't know where to find the correct value, see [how to use existing Azure AD app in TeamsFx project](https://github.com/OfficeDev/TeamsFx/blob/dev/docs/fx-core/using-existing-aad.md):
-
-```json
-"$schema": "https://aka.ms/teamsfx-env-config-schema",
-"description": "...",
-"manifest": {
-  ...
-},
-// Add code below. Note you need to replace the placeholders with real values.
-"auth": {
-    "clientId": "<your Azure AD app client id>",
-    "clientSecret": "{{$env.ENV_NAME_THAT_STORES_YOUR_SECRET}}",
-    "objectId": "<your Azure AD app object id>",
-    "accessAsUserScopeId": "<id of the access_as_user scope>"
-}
-```
+#### Use an existing Azure AD app for your bot
 
-After adding the snippet, add your client secret to the related environment variable for Teams Toolkit to resolve the actual client secret during provision.
+You can follow below steps to add environment variables to the .env files to use an Azure AD app created for your Teams app. If you don't have an Azure AD app for your bot yet or you already have one but don't know where to find the correct values, see [how to use existing Azure AD app in TeamsFx project](https://github.com/devdiv-azure-service-dmitryr/teamsfx-docs/blob/main/V5-doc-update/use-existing-aad-app.md).
```

Comment:
Change the linke of `how to use existing Azure AD app in TeamsFx project` to https://review.learn.microsoft.com/en-us/microsoftteams/platform/toolkit/use-existing-aad-app?branch=pr-en-us-8531

Reply:
Updated.